specs/core/0_beacon-chain.md

```diff
@@ -1003,11 +1057,39 @@ If `last_state_recalculation_slot % POW_RECEIPT_ROOT_VOTING_PERIOD == 0`, then:
 * If for any `x` in `state.candidate_pow_receipt_root`,  `x.votes * 2 >= POW_RECEIPT_ROOT_VOTING_PERIOD` set `state.processed_pow_receipt_root = x.receipt_root`.
 * Set `state.candidate_pow_receipt_roots = []`.
 
-### Validator set change
+#### Proposer reshuffling
```

Comment:
Why did we move this here?

Reply:
Aah, I think my goal was to move it out of the then-overloaded "Finally" section. We could move it to happen right before Finally if desired. I'm not sure what order is best.

Reply:
The order doesn't particularly matter (I think). 

The only reason I noticed was -- if we have it before the change validator section then it might produce some `ShardReassignmentRecord`s that immediately get wiped out when removing validators in `change_validators`.

Again... not a pressing concern